python/ml4ir/applications/ranking/model/losses/listwise_losses.py

```diff
@@ -26,18 +26,28 @@ def get_loss_fn(self, **kwargs):
         mask = kwargs.get("mask")
 
         def _loss_fn(y_true, y_pred):
+            """
+            Shapes
+            ------
+            y_true : [batch_size, num_classes, 1]
+            y_pred : [batch_size, num_classes, 1]
+            mask : [batch_size, num_classes, 1]
+            """
             batch_size = tf.cast(tf.shape(y_true)[0], tf.float32)
 
             # Mask the padded records
             y_true = tf.gather_nd(y_true, tf.where(tf.equal(mask, tf.constant(1.0))))
             y_pred = tf.gather_nd(y_pred, tf.where(tf.equal(mask, tf.constant(1.0))))
 
-            # Reshape the tensors
-            y_true = tf.expand_dims(tf.squeeze(y_true), axis=-1)
-            y_pred = tf.expand_dims(tf.squeeze(y_pred), axis=-1)
+            # Reshape the tensors so that we sum the losses from each record
+            y_true = tf.expand_dims(y_true, axis=-1)
+            y_pred = tf.expand_dims(y_pred, axis=-1)
 
+            # Scale the sum of losses down by number of queries in the batch
             return tf.math.divide(bce(y_true, y_pred), batch_size)
 
+            return bce(y_true, y_pred, sample_weight=mask)
```

Comment:
there seems to be 2 return statements here, same identatation, no if/else. Typo?

Reply:
Good catch. Forgot the delete the older one below. I've been playing with a bunch of different ways to do this. Using the `sample_weight` argument is a little more difficult than I expected.